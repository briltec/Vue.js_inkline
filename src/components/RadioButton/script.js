import IButton from 'inkline/components/Button';
import IRadio from 'inkline/components/Radio';

export default {
    name: 'RadioButton',
    components: {
        IButton
    },
    data() {
        return {
            parentFormGroupName: 'IRadioButtonGroup'
        }
    },
    extends: IRadio
};
