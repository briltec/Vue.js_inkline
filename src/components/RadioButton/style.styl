@import '../../resources/css/config'


// @TODO
// Use actual button markup for radio buttons
//
.radio.button
    position: relative
    display: inline-block
    outline: none

    > input[type='radio']
        opacity: 0
        outline: none
        position: absolute

        &:checked + .inner
            color: white
            background-color: #409eff
            border-color: #409eff
            box-shadow: -1px 0 0 0 #409eff

    &:first-child > .inner
        border-left: 1px solid #dcdfe6
        border-radius: 4px 0 0 4px
        box-shadow: none!important

    &:last-child > .inner
        border-radius: 0 4px 4px 0

    &.-disabled
        color: --text-muted

    > .inner
        position: relative
        display: inline-block
        outline: none
        white-space: nowrap
        vertical-align: middle
        background: #fff
        border: 1px solid #dcdfe6
        border-left: 0
        -webkit-appearance: none
        box-sizing: border-box
        cursor: pointer
        transition: all .3s cubic-bezier(.645,.045,.355,1)
        padding: --spacers['3-4'] --spacers['2']
        border-radius: 0
        font-size: --font-size
