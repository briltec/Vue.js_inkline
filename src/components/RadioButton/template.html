<label class="radio button" :class="classes">
    <input
        type="radio"
        v-model="model"
        :value="currentValue"
        :disabled="isDisabled">
    <span class="inner">
        <slot></slot>
    </span>
</label>
