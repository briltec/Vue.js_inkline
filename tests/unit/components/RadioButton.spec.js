import { mount } from '@vue/test-utils';
import RadioButton from 'inkline/components/RadioButton';

describe('Components', () => {
    describe('RadioButton', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(RadioButton);
        });

        test('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
