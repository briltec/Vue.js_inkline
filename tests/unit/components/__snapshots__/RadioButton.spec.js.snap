// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components RadioButton should render correctly 1`] = `<i-button-stub activeclass="-active" tag="button" size="" tabindex="1" variant="" class="-radio -custom"><input type="radio" name="" value=""> </i-button-stub>`;
